Index: update-manager-0.200.6+canaima0/po/POTFILES.in
===================================================================
--- update-manager-0.200.6+canaima0.orig/po/POTFILES.in	2011-12-23 13:18:30.145636926 -0430
+++ update-manager-0.200.6+canaima0/po/POTFILES.in	2011-12-23 13:18:53.689641529 -0430
@@ -30,6 +30,8 @@
 UpdateManager/DistSpecific/changelog.py
 UpdateManager/DistSpecific/Debian/changelog.py
 UpdateManager/DistSpecific/Debian/__init__.py
+UpdateManager/DistSpecific/Canaima/changelog.py
+UpdateManager/DistSpecific/Canaima/__init__.py
 UpdateManager/DistSpecific/__init__.py
 UpdateManager/DistSpecific/loader.py
 UpdateManager/__init__.py
