Index: update-manager-0.200.6+canaima0/setup.py
===================================================================
--- update-manager-0.200.6+canaima0.orig/setup.py	2011-12-23 13:19:18.458141850 -0430
+++ update-manager-0.200.6+canaima0/setup.py	2011-12-23 13:19:43.409781681 -0430
@@ -24,6 +24,7 @@
                 'UpdateManager.DistSpecific',
                 'UpdateManager.DistSpecific.Debian',
                 'UpdateManager.DistSpecific.Ubuntu',
+		'UpdateManager.DistSpecific.Canaima',
                 'UpdateManager.Util'
                 ],
       package_dir={
