Index: update-manager/UpdateManager/DistSpecific/Canaima/changelog.py
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ update-manager/UpdateManager/DistSpecific/Canaima/changelog.py	2013-09-09 13:54:51.968257130 -0430
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+# UpdateManager/DistSpecific/Debian/changelog.py
+#
+#  Copyright (c) 2011 Equipo de Desarrollo de Canaima GNU/Linux 
+#                <desarrolladores@canaima.softwarelibre.gob.ve>
+#
+#  Author: Luis Alejandro Martínez Faneyth <martinez.faneyth@gmail.com>
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License as
+#  published by the Free Software Foundation; either version 2 of the
+#  License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+#  USA.
+
+""" Canaima-specific functionality. """
+
+import logging
+
+LOG = logging.getLogger('UpdateManager.DistSpecific.Canaima.changelog')
+
+from UpdateManager.DistSpecific.changelog import HTTPChangelogFetcher
+
+class CanaimaChangelogFetcher(HTTPChangelogFetcher):
+    """ Canaima-specific ChangelogFetcher implementation. """
+    def _get_changelog_url(self, pkg_info):
+        """ Debian-specific changelog URL generation. """
+
+        candidate_version = pkg_info.get_candidate_version()
+        srcpkg_name = pkg_info.get_source_package_name()
+        if ':' in candidate_version:
+            candidate_version = candidate_version[
+                candidate_version.find(':')+1:]
+        url = "http://packages.debian.org/changelogs/pool/"
+        candidate_uri = pkg_info.get_candidate_uri().split('/pool/', 1)[1]
+        candidate_uri = candidate_uri[:candidate_uri.rfind('/')]
+        url += '%s/%s_%s/changelog.txt' % (candidate_uri, srcpkg_name,
+                                           candidate_version)
+        LOG.debug("Downloading changelog for %s from %s.", srcpkg_name,
+                  url)
+        return url
Index: update-manager/UpdateManager/DistSpecific/Canaima/__init__.py
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ update-manager/UpdateManager/DistSpecific/Canaima/__init__.py	2013-09-09 13:54:51.912257127 -0430
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+# UpdateManager/DistSpecific/Canaima/__init__.py
+#
+#  Copyright (c) 2011 Equipo de Desarrollo de Canaima GNU/Linux
+#                <desarrolladores@canaima.softwarelibre.gob.ve>
+#
+#  Author: Luis Alejandro Martínez Faneyth <martinez.faneyth@gmail.com>
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License as
+#  published by the Free Software Foundation; either version 2 of the
+#  License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+#  USA.
+
+import logging
+
+from UpdateManager.DistSpecific import DistBase, UPDATE_CATEGORY
+from UpdateManager.DistSpecific.Canaima.changelog import CanaimaChangelogFetcher
+
+LOG = logging.getLogger('UpdateManager.Backend.Canaima')
+
+class CanaimaDist(DistBase):
+
+    def __init__(self):
+        DistBase.__init__(self, name="Canaima GNU/Linux",
+                          changelog_fetcher=CanaimaChangelogFetcher,
+                          distupgrade_check=False)
+
+    def get_update_category(self, pkg_info):
+        """ Returns the update category for a given package """
+        # We detect the update category using the
+        # origin's label, archive name and whether its
+        # trusted or not.
+        label = pkg_info.get_candidate_origin_label()
+        origin = pkg_info.get_candidate_origin_name()
+        trusted = pkg_info.candidate_origin_is_trusted()
+
+        # Untrusted updates are most likely unofficial ones, so
+        # we only need to check whether the source is trusted first.
+        if not trusted:
+            return UPDATE_CATEGORY.THIRDPARTY
+
+        if label == "Seguridad" and origin == "Canaima":
+            return UPDATE_CATEGORY.SECURITY
+
+        if label == "Backports.org archive" and \
+           origin == "Backports.org archive":
+            return UPDATE_CATEGORY.BACKPORT
+
+        # TODO: how to best detect backports?
+
+        elif label == "Canaima" and origin == "Canaima":
+            return UPDATE_CATEGORY.DEFAULT
+
+        # All other sources are third-party
+        LOG.debug('Could not identify update: origin=%s,label=%s,trusted=%s',
+                  origin, label, trusted)
+        return UPDATE_CATEGORY.THIRDPARTY
+
+    def get_bug_script_name(self):
+        return "debian_reportbug.sh"
